src/java/org/apache/cassandra/index/sai/QueryContext.java

```diff
@@ -66,9 +67,10 @@ public class QueryContext
 
     private final LongAdder shadowedPrimaryKeyCount = new LongAdder();
 
-    // Determines the order of using indexes for filtering and sorting.
-    // Null means the query execution order hasn't been decided yet.
-    private FilterSortOrder filterSortOrder = null;
+    private Plan originalPlan = null;
+
+
+    private Plan optimizedPlan = null;
```

Comment:
Do we need some defensive null safety? Safe defaults in case the plan is not set?

Reply:
Good idea

Reply:
Fixed